### ISSUE TITLE
Topic Analytics

### ISSUE THREAD

**Issue by averagewifiuser** [testora-app/testora-api#1]
### Action Items
- [x] topic they're doing well in for each subject
- [x] topic they're not doing well in for each subject
- [x] recommend study topics based on average worst topic for each subject
[split into highly, moderately, low recommended]

### Implementation Notes
** Just build it as classes and functions, we would worry about the implementation on thread later. so write them to run without needing app context**
they would see a break down for test level
-- topics in each test and the percentage they got

-- so frontend will add view analytics

would require a table for tracking that for each subject
-- rely on history of it as well

we need a way to get all the score average for the topics
-- basically...which topics are the getting very high scores at and low scores at

either run this when you're marking.... or run it for each test but that's more accurate...
1. questions correct per topic
2. questions wrong per topic
